yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -447,14 +447,16 @@ export class AztecNodeService implements AztecNode, AztecNodeAdmin, Traceable {
     return this.logsSource.getPrivateLogs(from, limit);
   }
 
-  /**
-   * Gets all logs that match any of the received tags (i.e. logs with their first field equal to a tag).
-   * @param tags - The tags to filter the logs by.
-   * @returns For each received tag, an array of matching logs is returned. An empty array implies no logs match
-   * that tag.
-   */
-  public getLogsByTags(tags: Fr[]): Promise<TxScopedL2Log[][]> {
-    return this.logsSource.getLogsByTags(tags);
+  public async getPrivateLogsByTags(tags: Fr[]): Promise<TxScopedL2Log[][]> {
+    const allLogs = await this.logsSource.getLogsByTags(tags);
+    // TODO(#14460): Have logSource implement getPrivateLogsByTags and skip the filter here.
```

Comment:
I could do this issue straight away if desired. It's not blocking anything now though so is probably too low of a priority.

Reply:
I'd first wait until we finalize our API needs and then do the changes in one go, since for the node to be efficient at this it may require changes to the archiver etc.

Reply:
Hm doing this here would create a breaking change in the node API. I'd try to minimize that and only do it once we know what we're going towards for sure. What if we implemented this change in the client? So we keep calling `getLogsByTags`, but filter locally once we've got the data. We could then return two different types for private and public instead of the wider 'tx scoped l2 log'

Reply:
The least complicated way for me to keep the backwards compatibility is to just keep also the old `getLogsByTags` function on the `AztecNode` API. That would mean that updated PXE would need an updated node but that should be fine as we are worried about old PXEs not being compatible with new nodes, rigth?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/14459/commits/2d318b363423dadc6cd342be1de104695706c6ae